typespec/Applications.Core/environments.tsp

```diff
@@ -110,11 +113,14 @@ model SecretConfig {
   secret?: string;
 }
 
-// ProviderConfigProperties allows to get the additional properties. To ensure that `additionalProperties` is true, we need to extend `Record<unknown>`.
+// ProviderConfigProperties allows to get the additional properties and secrets. To ensure that `additionalProperties` is true, we need to extend `Record<unknown>`.
 // Reference: https://github.com/Azure/typespec-azure/blob/main/packages/typespec-autorest/test/additional-properties.test.ts
 #suppress "@azure-tools/typespec-azure-core/bad-record-type"
 @doc("This configuration holds the necessary information to authenticate and interact with a provider for the recipe execution.")
-model ProviderConfigProperties extends Record<unknown> {}
+model ProviderConfigProperties extends Record<unknown> {
+  @doc("Sensitive data for the referenced resource stored as secrets. The secrets are stored in Applications.Core/SecretStores resource.")
```

Comment:
What is "referenced resource" in this context?

Reply:
updated comment to "Sensitive data in provider configuration can be stored as secrets..."